src/GraphQL/Execution/DocumentExecuter.cs

```diff
@@ -204,12 +196,6 @@ private async Task<ExecutionResult> CoreExecuteAsync(ExecutionOptions options)
                         }).ConfigureAwait(false);
                 }
 
-                if (options.ComplexityConfiguration != null && validationResult.IsValid && analyzeComplexity)
-                {
-                    using (metrics.Subject("document", "Analyzing complexity"))
```

Comment:
@Shane32 What do you think about exposing metrics to all validation rules? Not a whole `Metrics` instance but some sort of method to delegate control flow to `public Marker Subject(string category, string? subject, Dictionary<string, object?>? metadata = null)`. Then we may preserve complexity time metrics from the removed code.

Reply:
Or add `IMetrics` interface with single method `Marker Subject(string category, string? subject, Dictionary<string, object?>? metadata = null)` and add `IMetrics Metrics { get; set; }` property into `ValidationContext`.

Reply:
#3176